ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -122,10 +125,11 @@ void *libX11_Variadic_u64(uint64_t a_0, size_t count, unsigned long *list, void
       # x1 = <count>
       # x9 = <list ptr>
 
-      # Stack objects
-      # Count >= 7
+      # The number of arguments on the stack will always be (Count - 7) + 1
+
```

Comment:
The comment can be simplified a lot.

I would add this under `.stack%=:`:
```suggestion
      # The first 7 variadic arguments fit into reserved registers, but we must copy the rest to the stack. X11 further requires us to add a terminating nullptr, so in total (count - 6) arguments are written to the stack, starting from list[7].
```

Then this entire block code section can be replaced with:
```
      # In x1, store the number of arguments written to stack (including the terminating nullptr).
      # We must round to the nearest pair because TODO
      sub x1, x1, 6
      and x10, x1, 1
      add x10, x10, x1
```

The top comment then also explains why we skip the first 7 elements in x12 below.